packages/vitest/src/node/plugins/index.ts

```diff
@@ -233,22 +233,26 @@ export async function VitestPlugin(
 
         hijackVitePluginInject(viteConfig)
       },
-      async configureServer(server) {
-        if (options.watch && process.env.VITE_TEST_WATCHER_DEBUG) {
-          server.watcher.on('ready', () => {
-            // eslint-disable-next-line no-console
-            console.log('[debug] watcher is ready')
-          })
-        }
-        await ctx.setServer(options, server, userConfig)
-        if (options.api && options.watch) {
-          (await import('../../api/setup')).setup(ctx)
-        }
+      configureServer: {
+        // runs after vite:import-analysis as it relies on `server` instance on Vite 5
+        order: 'post',
+        async handler(server) {
```

Comment:
Btw, I also tried moving it to "post middleware" hook like below, but it turns out Vite doesn't await returned promise so it didn't work https://github.com/vitejs/vite/blob/95020ab49e12d143262859e095025cf02423c1d9/packages/vite/src/node/server/index.ts#L881-L884

```sh
{
  configureServer(server) {
    return async () => {
      await ctx.setServer(options, server, userConfig)
      ...
    }
  }
}
```